### DISCUSSION TITLE
Stream Deck SDK "Getting Started" instructions problem: symlinks don't seem to work on Mac?

### DISCUSSION THREAD

**Discussion by woodrow** [elgatosf/elgato#63] [Help] [answered]
Did the Stream Deck app change to stop following symlinks? I've been trying to follow the instructions here https://docs.elgato.com/sdk/plugins/getting-started#macos-symlink, including trying absolute paths for my symlink [1] rather than relative paths, but I can't get the plugin to load. I can only successfully get my new plugin to register when I actually `cp` the template plugin directory into `~/Library/Application Support/com.elgato.StreamDeck/Plugins`.

Can you please either update your SDK docs or provide guidance about the current best approach for Stream Deck plugin development?

Context on my setup:
- macOS version 13.4 (22F66)
- Elgato Stream Deck.app version 6.3.0 (18948)
- streamdeck-plugin-template HEAD 38fcbfa7a558523ca0d182e8b3b30d95fd02b865 (origin/main)

[1] inspired by discussion here https://github.com/elgatosf/streamdeck-plugin-template/issues/8

**Comment by GeekyEggo** [comment]
Hey @woodrow, could I ask what the name of the folder that's being sym-linked is called?

As part of loading plugins that are sym-linked, it's required the folder follow a pre-defined name structure, specifically a reverse-DNS name followed by `.sdPlugin`, for example:
- `com.richard.test.sdPlugin`
- `com.elgato.wave-link.sdPlugin`

Having reviewed the [template's README](https://github.com/elgatosf/streamdeck-plugin-template), I can see this requirement isn't made clear, my apologies. I will ensure it is updated.

**Reply by woodrow** [reply]
Here's an example of a symlink to the [demo counter plugin](https://github.com/elgatosf/streamdeck-counter) that doesn't work for me, even though it seems to follow the naming convention:

```sh-session
$ ls -l ~/Library/Application\ Support/com.elgato.StreamDeck/Plugins/com.elgato.counter.sdPlugin
lrwxr-xr-x  1 woodrow  staff  75 Aug 26 17:28 /Users/woodrow/Library/Application Support/com.elgato.StreamDeck/Plugins/com.elgato.counter.sdPlugin -> /Users/woodrow/proj/streamdeck-counter/Sources/com.elgato.counter.sdPlugin/
```

**Comment by neilkimmett** [comment]
+1 to this, spent a lot of time trying to get symlink to work but couldn't; ended up just copying the folder in.

**Reply by ChekTek** [reply, marked as answer]
Hi @neilkimmett 

We are working on a new Node.js SDK beta and a CLI tool. If you are familiar with NPM, you may have better luck using the CLI tool command `streamdeck link` in your `.sdPlugin` folder to handle the symlink into Stream Deck. 

https://www.npmjs.com/package/@elgato/cli

There are also some issues with some symlink tools using relative paths, which could cause issues with Stream Deck.

**Reply by neilkimmett** [reply]
Oh this looks great, thank you!
